src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -353,35 +340,64 @@ public override void Flush()
             Interop.Termios.TermiosDiscard(_handle, Interop.Termios.Queue.AllQueues);
         }
 
-        public override long Seek(long offset, SeekOrigin origin)
+        internal int ReadByte(int timeout)
         {
-            throw new NotSupportedException(SR.NotSupported_UnseekableStream);
+            Read(_tempBuf, 0, 1, timeout);
+            return _tempBuf[0];
         }
 
-        public override void SetLength(long value)
+        public override int Read(byte[] array, int offset, int count)
         {
-            throw new NotSupportedException(SR.NotSupported_UnseekableStream);
+            return Read(array, offset, count, ReadTimeout);
         }
 
-        public override int ReadByte()
+        internal unsafe int Read(byte[] array, int offset, int count, int timeout)
         {
-            return ReadByte(ReadTimeout);
+            Task<int> t = ReadAsync(array, offset, count, GetCancellationTokenFromTimeout(timeout));
```

Comment:
GetCancellationTokenFromTimeout is creating a new CancellationTokenSource, which is backed by a timer.  The most common case is the operation will complete before the timer fires, in which case we don't want the Timer and its resources to sit around even though they're no longer needed or useful.  Thus, the CTS that's created should be disposed of once the associated operation completes.